mangaki/mangaki/tests/test_posters.py

```diff
@@ -0,0 +1,61 @@
+from unittest.mock import patch, MagicMock, PropertyMock
+
+from django.test import TestCase
+
+from mangaki.models import Editor, Studio, Work, Category
+
+from mangaki.utils.db import get_potential_posters
+
+
+class PostersTest(TestCase):
+
+    def setUp(self):
+        # FIXME: The defaults for editor and studio in Work requires those to
+        # exist, or else foreign key constraints fail.
+        Editor.objects.create(pk=1)
+        Studio.objects.create(pk=1)
+
+        anime = Category.objects.get(slug='anime')
+
+        self.kiznaiver = Work.objects.create(
+            title='Kiznaiver',
+            category=anime,
+            ext_poster='bRoKeN_lInK'  # That's how I feel when I see a broken poster.
+        )
+
+    # We only want to abstract the MAL search implementation.
+    @patch('mangaki.utils.db.client.search_work')
+    def test_get_potential_posters(self, mocked_search):
+        with self.subTest('When MAL returns no poster'):
+            expected = [{
+                'current': True,
+                'url': self.kiznaiver.ext_poster
+            }]
+            # Set up mocks.
+            mocked_entry = MagicMock()
+            # Pay no mind to this ugliness:
+            # https://docs.python.org/3/library/unittest.mock.html#unittest.mock.PropertyMock
+            type(mocked_entry).poster = PropertyMock(return_value=None)
+            mocked_search.return_value = mocked_entry
+            # Let the magic occur.
+            posters = get_potential_posters(self.kiznaiver)
+            # In this case, `get_potential_posters` cannot fix the current poster.
+            self.assertCountEqual(posters, expected)
```

Comment:
`assertEqual(Counter(list(posters)), Counter(list(expected)))` and `assertEqual(sorted(posters), sorted(expected))` are doing exactly the same thing, except that the error messages in the first case can be more explicit.

It works with unhashable object because it "does the same thing as", it is not implemented as is. Mock implementation off the top of my head:

```python
def assertCountEqual(first, second):
    items, first_counts = [], []
    for first_el in first:
        try:
            first_counts[items.index(first_el)] += 1
        except ValueError:
            items.append(first_el)
            first_counts.append(1)
    second_counts = [0 for _ in first_counts]
    for second_el in second:
        try:
            second_counts[items.index(second_el)] += 1
        except ValueError:
            raise ValueError("%s is present in second but not in first." % second_el)
    for el, first_count, second_count in zip(items, first_counts, second_counts):
        if first_count != second_count:
            raise ValueError("%s is present %d times in first but %d times in second." % (el, first_count, second_count))
```

This works with non-hashable objects because `.index()` uses `==` comparison.